kamaki/clients/compute/__init__.py

```diff
@@ -441,3 +441,56 @@ def delete_floating_ip(self, tenant_id, fip_id=None):
         """
         r = self.floating_ips_delete(tenant_id, fip_id)
         return r.headers
+
+    def list_keypairs(self):
+        """
+        :returns: (list) [
+            {id: ..., name: ..., public_key: ..., fingerprint: ...},
+            ... ]
+        """
+        r = self.keypairs_get()
+        return r.json['keypairs']
+
+    def get_keypair_details(self, key_name=None):
+        """
+        :param key_name: (str) The name of the key
+
+        :returns: (dict) {keypair: ..., id: ..., name: ..., public_key: ...,
+                                   fingerprint: ...., created_at: ...,
+                                   updated_at: ...}
+        """
+        r = self.keypairs_get(key_name=key_name)
+        return r.json['keypair']
+
+    def create_or_update_key(self, key_name=None, public_key=None):
```

Comment:
I suggest key_name should never be None.

When we decide to implement the "auto generate" feature, we can do this in CLI.